library/ssl_cli.c

```diff
@@ -642,7 +637,8 @@ static void ssl_write_alpn_ext( mbedtls_ssl_context *ssl,
     /* Skip writing extension and list length for now */
     p += 4;
 
-    for( cur = ssl->conf->alpn_list; *cur != NULL; cur++ )
+    for( cur = ssl->conf->alpn_list;
+	 p < buf + 6 + alpnlen && cur != NULL; cur++ )
```

Comment:
Thanks @andresag01, I've given it another thought and think it's inconsistent to silently cut an overly long list here while failing on similar occasions in the rest of the code. I'll change that.